### PR TITLE
[RISCV] Don't combine store of vmv.x.s/vfmv.f.s to vp_store with VL of 1 when it's indexed store

### PR DESCRIPTION
**Pull request by zixuan-wu**
Because we can't support vp_store with indexed address mode by lowering to vse intrinsic later.

### CODE REVIEW

File: llvm/test/CodeGen/RISCV/rvv/combine-store-extract-crash.ll [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  ret void
+}
+
+; Function Attrs: nocallback nofree nosync nounwind speculatable willreturn memory(none)
+declare i32 @llvm.vector.reduce.add.v16i32(<16 x i32>) #1
+
+; uselistorder directives
+uselistorder ptr @llvm.vector.reduce.add.v16i32, { 1, 0 }
+
+attributes #0 = { "target-features"="+a,+c,+d,+f,+m,+relax,+v,+xtheadba,+xtheadbb,+xtheadbs,+xtheadcmo,+xtheadcondmov,+xtheadfmemidx,+xtheadmac,+xtheadmemidx,+xtheadmempair,+xtheadsync,+xtheadvdot,+zba,+zbb,+zbc,+zbs,+zfh,+zicsr,+zihintpause,+zve32f,+zve32x,+zve64d,+zve64f,+zve64x,+zvl128b,+zvl32b,+zvl64b" }
```
Comment (wangpc-pp) [inline llvm/test/CodeGen/RISCV/rvv/combine-store-extract-crash.ll:77]: These attributes and metadatas can be removed.

Comment (zixuan-wu) [reply llvm/test/CodeGen/RISCV/rvv/combine-store-extract-crash.ll:77]: The assert can't reproduce if tbaa info or some features are removed.

Comment (lukel97) [reply llvm/test/CodeGen/RISCV/rvv/combine-store-extract-crash.ll:77]: Target features are usually passed in the llc flags

File: llvm/test/CodeGen/RISCV/rvv/combine-store-extract-crash.ll [hunk 2]
```diff
@@ -0,0 +1,78 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 2
+; RUN: llc -mtriple=riscv32 -mattr=+v,+xtheadmemidx,+xtheadmempair -verify-machineinstrs < %s \
```
[the remark this answers is not in the document]
Comment (topperc) [reply llvm/test/CodeGen/RISCV/rvv/combine-store-extract-crash.ll:2]: Yes. Indexed scalar loads are only legal with T-Head extensions.

**Review by topperc — APPROVED** [review]
LGTM

**Pull request merged**